apps/ejabberd/src/mod_mam_muc_odbc_async_pool_writer.erl

```diff
@@ -43,7 +43,7 @@
                 max_packet_size     :: non_neg_integer(),
                 max_subscribers     :: non_neg_integer(),
                 host                :: ejabberd:server(),
-                conn,
+                connection_pool     :: atom(),
```

Comment:
Since the number of connections in the pool is equal to the number of workers the queue time is small compared to the query time (I measured!). In MAM tests, the average time for "checkout" (48 μs) takes only ~10% of the average total time for query (410 μs). When under load in real-life scenario the ratio will only get smaller as the batch flushes get bigger.